persistence-modules/spring-boot-persistence-2/src/main/java/com/baeldung/db/indexing/FileSystemRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.db.indexing;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+interface FileSystemRepository extends JpaRepository<Image, Long> {
+
+    String RESOURCES_DIR = FileSystemRepository.class.getResource("/")
+        .getPath() + "/";
+
+    default Image saveToFileSystem(Image image) throws Exception {
+        Path newFile = Paths.get(RESOURCES_DIR + new Date().getTime() + "-" + image.name);
+        Files.createDirectories(newFile.getParent());
+
+        Files.write(newFile, image.content);
+
+        image.setName(image.name);
```

Comment:
Why are you setting the image name to itself? and why do you have both a setter and a public property on `Image`? I'd rather than we just used getters and setters?